lib/xcodeproj/project/object/helpers/file_references_factory.rb

```diff
@@ -180,15 +180,16 @@ def new_subproject(group, path, source_tree)
             ref = new_file_reference(group, path, source_tree)
             ref.include_in_index = nil
 
-            product_group_ref = find_products_group_ref(group, true)
+            product_group_ref = group.project.new(PBXGroup)
```

Comment:
should we create a new group each time?

Reply:
Waiting to hear back from a team member regarding our Xcode set up. Will get back to you as soon as I know.

Reply:
It works the same for us. The concern we had with having this be a new group was that our subprojects would generate as a new subproject folder instead of a new subproject xcodeproj file in the navigator. I just ran through this with our main engineer that set up our Xcode project and verified that it still sets up a new <subproject>.xcodeproj file as we want.

As far as side effects however, I don't believe him or I can really speak beyond just our project. If you have any other people that are very familiar with Xcode that want to take a look at this I'd be happy to see the input and correct anything that's wrong, but that's out of my expertise at that point.

I just know that this fixes our crashing when trying to change our project settings. 

Happy to hear any concerns and implement any changes.

Also, I found this article that describes a much more verbose fix for the same issue. Take a look and compare and let me know what you think, perhaps their solution is better? https://www.programmersought.com/article/83996170465/

Reply:
Any feedback? Sorry we can't be of more help.

Reply:
We appreciate your time. Totally understand not wanting to ship a breaking change.

Reply:
Hey there, don't mean to rush you if you're busy, just checking in if you had any updated thoughts on this.

Reply:
Let us know if you'll find the time to get to this anytime soon please @dnkoutso 🙂 If not it make more sense for us to fork and just wait for this to be fixed before reverting back to using the official version.

Reply:
Sorry for the long delay, I recommend for now to fork and proceed. I have not had time to review this and evaluate it. I apologize but normal day-to-day work stuff................

Reply:
No worries, totally get it. We will patch and host it locally for now, and when you get to this and check it out, it would be great if you could let us know so we can revert back to the public one. :)

Reply:
Just FYI, after doing more of our own internal testing it looks like this ends up breaking one of the entries in our linking phase.

Reply:
thats "good" to hear I guess! Without some deeper investigation I would be afraid to land this. Again really sorry for lack of traction here, the repercussions can be large.

Reply:
Yeah, totally. Wouldn't want to break anything for anyone. 

Looking around though, it does seem like this has been an issue for multiple people. Are there any plans to investigate this further? Or, perhaps, to write up a guide on how to properly create subprojects so that we can see if perhaps we are mis-using the library?